azerothcore/azerothcore-wotlk
# fix(Scripts/HyjalSummit): Rewrite Battle for Mount Hyjal

restore the PR https://github.com/azerothcore/azerothcore-wotlk/pull/18460

the real author is @Gultask 



## Changes Proposed:

This PR proposes changes to:

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4172
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4171
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4170
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/17214
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/17207
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/17215
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/17216

## SOURCE:

The changes have been validated through:

A lot has been provided by the CMangos team, mainly wave positions.
Pathing has been sniffed by me for the most part.

## Tests Performed:

This PR has been:


## How to Test the Changes:



1. .ins u all
2. .tele hyjalsummit
3.

## Known Issues and TODO List:


- [x] If summons (totems, spirit wolf, water elementals) are up when a reset happens, the worldserver crashes // It stopped?
- [x] Crash on player death
- [x] General crash prevention
- [x] Thrall, Tyrande
- [x] Trash scripting
- [x] Trash do not move past the first point of their second path (PathEndReached())
- [x] Boss modernization
- [x] DB rework for bases
- [x] Scourge Base
- [x] Archimondo (RP Yell after Azgalor, Spawning in correctly, Spawn after crash/wipe/reset)
- [x] DB Cleanup
- [x] Infernals


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

Co-authored-by: Gultask <sagemochi@hotmail.com>
